content/js1/sprints/1/prep/index.md

```diff
@@ -8,4 +8,443 @@ backlog= 'Module-JS1'
+
+Instead we make use of **interfaces**.
+
+{{<note title="Definition" type="info">}}
+
+An **interface** can be thought of as a boundary that allows communication between a user and a machine.
+The user asks the machine to carry out certain tasks via the interface. We can think of the interface as a gateway between the user and the machine.
+
+{{</note>}}
+
+For example, we may want to withdraw money from an ATM machine. We go to a machine in the wall with a screen and a keypad. The **screen and the keypad** form the user interface in this case. We can use this user interface to ask the machine to carry out certain tasks - like give my balance, or please withdraw some money from my account.
+
+{{<note title="Activity" type="info">}}
+
+Try and define the **user interface** for the following devices:
+
+- a calculator
+- a microwave
+- a desktop lamp
+- Alexa
+- ChatGPT
+
+{{</note>}}
+
+### Terminal interface
+
+As with other machines, programmers need to use interfaces to get machines to carry out tasks.
+
+A computer terminal is an **interface** where programmers can issue commands to a computer. On a Mac, the terminal looks like this:
+
+![terminal](terminal.png)
+
+The terminal appears as a window on the computer, prompting users for instructions. So we can input a command into the prompt and hit enter. After hitting enter, the interface passes this command along for the computer to execute. For example, we can input the `ls` command into the terminal:
+
+![enter-ls-command](enter-ls-command.png)
+
+We can then press enter to issue this command to the computer and get a response:
+
+![ls-feedback](ls-feedback.png)
+
+As users enter text instructions and receive text output, we say that the terminal is a **text-based interface**.
+
+### Writing computer instructions
+
+We can issue commands to the computer using the terminal. These commands are instructions that the computer knows how to interpret. Given the `ls` command, the computer knows this is an instruction to list the files and directories in the current directory.
+
+However, we can make use of other programming languages too.
+A programming language is a particular set of rules for writing computer instructions.
+During the execution of a computer program, a computer will store and modify data.
+
+So we can think of a programming language as allowing us to create **data** and **operations** on that data.
+
+{{<note title="Definition" type="info">}}
+**Data** consists of information in different forms: for example, numbers and text could be different forms of data. The data in a executing program is sometimes referred to as the **state**.
+{{</note>}}
+
+{{<note title="Definition" type="info">}}
+**Operations** are anything that take data and modify existing data or create new data from the current data in the program.
+{{</note>}}
```

Comment:
I'm not sure this section flows very well? I think I'd restructure to something like this:

We can issue commands to the computer using the terminal. These commands are instructions that the computer knows how to interpret. Given the ls command, the computer knows this is an instruction to list the files and directories in the current directory.

During the execution of a computer program, a computer will store and data and modify it using operations

Definitions of data & operations...

`ls` in the terminal is an example of a programming language.

[Definition] A programming language is a particular set of rules for writing computer instructions.

We can use other programming languages too. In this course we're going to focus on the JavaScript programming language.